src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/DiscoverVariantsFromContigAlignmentsSAMSpark.java

```diff
@@ -167,21 +180,119 @@ public static AlignedContig parseReadsAndOptionallySplitGappedAlignments(final I
+
+        final List<VariantContext> impreciseVariants =
+                Utils.stream(evidenceTargetLinks)
+                        .filter(DiscoverVariantsFromContigAlignmentsSAMSpark::isImpreciseDeletion)
+                        .filter(e -> e._2.getReadPairs() + e._2.getSplitReads() > parameters.impreciseEvidenceVariantCallingThreshold)
+                        .map(e -> createImpreciseDeletionVariant(e._2, reference.getReferenceSequenceDictionary(null), reference))
+                        .collect(Collectors.toList());
 
-        SVVCFWriter.writeVCF(pipelineOptions, vcfFileName, fastaReference, annotatedVariants, toolLogger);
+        localLogger.info("Called " + impreciseVariants.size() + " imprecise deletion variants");
+        collectedAnnotatedVariants.addAll(impreciseVariants);
+        return collectedAnnotatedVariants;
+    }
+
+    private static VariantContext createImpreciseDeletionVariant(final EvidenceTargetLink e,
+                                                                 final SAMSequenceDictionary sequenceDictionary,
+                                                                 final ReferenceMultiSource reference) {
+        final SvType svType = new SimpleSVType.ImpreciseDeletion(e, sequenceDictionary);
+        return AnnotatedVariantProducer
+                .produceAnnotatedVcFromEvidenceTargetLink(e, svType, sequenceDictionary, reference);
+    }
+
+    private static VariantContext annotateWithImpreciseEvidenceLinks(final VariantContext variant,
+                                                                     final PairedStrandedIntervalTree<EvidenceTargetLink> evidenceTargetLinks,
+                                                                     final SAMSequenceDictionary referenceSequenceDictionary,
+                                                                     final ReadMetadata metadata,
+                                                                     final int defaultUncertainty) {
+        if (variant.getStructuralVariantType() == StructuralVariantType.DEL) {
+            SVContext svc = SVContext.of(variant);
+            final int padding = (metadata == null) ? defaultUncertainty : (metadata.getMaxMedianFragmentSize() / 2);
+            PairedStrandedIntervals svcIntervals = svc.getPairedStrandedIntervals(referenceSequenceDictionary, padding);
+
+            final Iterator<Tuple2<PairedStrandedIntervals, EvidenceTargetLink>> overlappers = evidenceTargetLinks.overlappers(svcIntervals);
+            int readPairs = 0;
+            int splitReads = 0;
+            while (overlappers.hasNext()) {
+                final Tuple2<PairedStrandedIntervals, EvidenceTargetLink> next = overlappers.next();
+                readPairs += next._2.getReadPairs();
+                splitReads += next._2.getSplitReads();
+                overlappers.remove();
```

Comment:
is this `remove()` necessary because you don't want to count an evidence twice for difference DEL records? Just to see if I'm understanding correctly.

Reply:
If a high-scoring evidence link cluster matches an assembly variant, we just want to annotate the assembly variant with the number of read pairs and split reads, but not call a new IMPRECISE variant. So here the `EvidenceTargetLink` gets used up and removed from the collection when used in annotation. The remaining `EvidenceTargetLink` objects that weren't used for annotation will then be considered for IMPRECISE variant calling downstream. And yes, this will also cause `EvidenceTargetLinks` from being used to annotate more than one variant.